ts/screens/wallet/TransactionsScreen.tsx

```diff
@@ -61,24 +52,40 @@ class TransactionsScreen extends React.Component<Props, never> {
     return (
       <WalletLayout
         title={I18n.t("wallet.paymentMethod")}
-        navigation={this.props.navigation}
         showPayButton={false}
         headerContents={headerContents}
-        cardType={{ type: CardEnum.FULL, card: this.props.selectedWallet }}
+        cardType={{ type: CardEnum.FULL, card: selectedWallet }}
+        navigateToWalletList={() =>
+          this.props.navigation.navigate(ROUTES.WALLET_LIST)
+        }
+        navigateToScanQrCode={() =>
+          this.props.navigation.navigate(ROUTES.PAYMENT_SCAN_QR_CODE)
+        }
+        navigateToWalletTransactions={(wallet: Wallet) =>
+          this.props.navigation.dispatch(
+            navigateToWalletTransactionsScreen({ selectedWallet: wallet })
+          )
+        }
       >
         <TransactionsList
           title={I18n.t("wallet.transactions")}
           totalAmount={I18n.t("wallet.total")}
-          navigation={this.props.navigation}
           display={TransactionsDisplayed.BY_WALLET}
+          navigateToTransactionDetails={transaction =>
+            this.props.navigation.dispatch(
+              navigateToTransactionDetailsScreen({
+                transaction,
```

Comment:
I still think is better to pass only the transactionId and delegate the loading of the complete transaction object to the screen.

Reply:
Why do you think it's better? Passing the id means that the transaction details screen needs to check that the transaction actually exist in the redux store (it may be passed a non-existent id) so that screen needs to deal with that case - instead here, the transaction details screen is sure to always have the info it needs and the logic is simpler.

Reply:
I see this approach as a message passed from a screen to another, providing the data needed for the user flow, this guarantees consistency and makes it easier to follow where the data is coming from. 

But I'm also curious what advantages do you see in passing just the id of the object? 

The only reason I can see for passing an id instead of the whole object is in the case the object may get modified or deleted by a concurrent process while the user is navigating from one screen to another - in that case yes the destination screen should load from the store the updated object and check if the object still exist. But that's not the case here.